tableau_utilities/tableau_file/tableau_file_objects.py

```diff
@@ -723,6 +723,11 @@ def dict(self):
 
 @dataclass
 class Refresh(TableauFileObject):
+    print(TableauFileObject)
+    print(type(TableauFileObject))
+    print('after printing the file object')
+
+    print(TableauFileObject.__dict__)
```

Comment:
I'm getting a strange error that I can't seem to debug.  I think it's because a key is missing in the code that parses the Tableau file but I'm having trouble looking in all these objects to figure it out.

The error is:
```
Traceback (most recent call last):
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/bin/tableau_utilities", line 33, in <module>
    sys.exit(load_entry_point('tableau-utilities==2.0.0', 'console_scripts', 'tableau_utilities')())
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/scripts/cli.py", line 254, in main
    args.func(args)
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/scripts/gen_config.py", line 380, in generate_config
    metadata_record_columns = get_metadata_record_columns(datasource_name, datasource_path, debugging_logs)
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/scripts/gen_config.py", line 94, in get_metadata_record_columns
    metadata_records = [c.dict() for c in Datasource(datasource_path).connection.metadata_records]
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/tableau_file/tableau_file.py", line 114, in __init__
    self.extract: tfo.Extract = tfo.Extract(**self.__get_section('extract')[0])
  File "<string>", line 9, in __init__
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/tableau_file/tableau_file_objects.py", line 863, in __post_init__
    self.connection = ParentConnection(**transform_tableau_object(self.connection))
  File "<string>", line 21, in __init__
  File "/Users/jayrosenthal/.pyenv/versions/3.10.3/lib/python3.10/site-packages/tableau_utilities/tableau_file/tableau_file_objects.py", line 767, in __post_init__
    self.refresh = Refresh(**transform_tableau_object(self.refresh))
TypeError: Refresh.__init__() got an unexpected keyword argument 'refresh_event'
```

I think that's erroring on reading this XML as that's the only place I see something with a refresh event.

```
      <extract _.fcp.ObjectModelExtractV2.true...object-id='' count='-1' enabled='true' units='records'>
        <connection access_mode='readonly' authentication='auth-none' author-locale='en_US' class='hyper' dbname='/Users/jayrosenthal/Documents/My Tableau Repository/Datasources/DIM_INFRASTRUCTURE_COSTS (INFRASTRUCTURE.DIM_INFRASTRUCTURE_COSTS) (INFRASTRUCTURE).hyper' default-settings='yes' schema='Extract' sslmode='' tablename='Extract' update-time='01/06/2023 12:00:35 AM' username='tableau_internal_user'>
          <_.fcp.ObjectModelEncapsulateLegacy.false...relation name='Extract' table='[Extract].[Extract]' type='table' />
          <_.fcp.ObjectModelEncapsulateLegacy.true...relation name='Extract' table='[Extract].[Extract]' type='table' />
          <refresh>
            <refresh-event add-from-file-path='DIM_INFRASTRUCTURE_COSTS (INFRASTRUCTURE.DIM_INFRASTRUCTURE_COSTS) (INFRASTRUCTURE)' increment-value='%null%' refresh-type='create' rows-inserted='44672386' timestamp-start='2023-01-05 23:38:02.848' />
          </refresh>
          <metadata-records>
            <metadata-record class='column'>
              <remote-name>INFRASTRUCTURE_PROVIDER</remote-name>
              <remote-type>129</remote-type>
              <local-name>[INFRASTRUCTURE_PROVIDER]</local-name>
```

I can't seem to figure out how to adjust (or if I'm in the right place) to fix. Any tips on approaching this?